metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -999,8 +1059,10 @@ void convert_batch_to_row_diff(const std::string &pred_succ_fprefix,
 
     async_writer.join();
 
-    set_rows_fwd.clear(); // free up memory
-    set_rows_bwd.clear();
+    if (swap_disk) {
+        set_rows_fwd.clear(); // free up memory
+        set_rows_bwd.clear();
+    }
```

Comment:
Unless I missed something, it looks like `set_rows_fwd` and `set_rows_bwd` are not used after this anyway, so why not clear them in all cases?

Reply:
Also, I'm struggling to see what these vectors actually do when in memory mode. It looks like you populate them and do nothing...

Reply:
it would be nice to have some comments describing the general flow to better understand this

Reply:
For in-memory transforms, these buffers store all the diffs, they are extracted in `call_diffs` (invoked below).
I added a comment.